OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1293,13 +1293,10 @@ int Model::getNumMuscleStates() const {
 int Model::getNumProbeStates() const {
 
     int n = 0;
-    for(int i=0;i<get_ProbeSet().getSize();i++){
-        Probe *p = dynamic_cast<Probe*>( &get_ProbeSet().get(i) );
```

Comment:
Original is bogus because the probeset contains `Probe`s, so it's a pointless `dynamic_cast`